product_pack/models/product_pack_line.py

```diff
@@ -0,0 +1,65 @@
+# Copyright 2019 Tecnativa - Ernesto Tejeda
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+from odoo import api, fields, models, _
+import odoo.addons.decimal_precision as dp
+from odoo.exceptions import ValidationError
+
+
+class ProductPackLine(models.Model):
+    _name = 'product.pack.line'
+    _description = 'Product pack line'
+    _rec_name = 'product_id'
+
+    parent_product_id = fields.Many2one(
+        'product.product',
+        'Parent Product',
+        ondelete='cascade',
+        index=True,
+        required=True,
+    )
+    quantity = fields.Float(
+        required=True,
+        default=1.0,
+        digits=dp.get_precision('Product UoS'),
+    )
+    product_id = fields.Many2one(
+        'product.product',
+        'Product',
+        ondelete='cascade',
+        index=True,
+        required=True,
+    )
+
+    # because on expand_pack_line we are searching for existing product, we
+    # need to enforce this condition
+    _sql_constraints = [
+        ('product_uniq', 'unique(parent_product_id, product_id)',
+         'Product must be only once on a pack!'),
+    ]
+
+    @api.constrains('product_id')
+    def check_relations(self):
```

Comment:
```suggestion
    def _check_relations(self):
```

Constraint methods are private always.